app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/ingredients_analysis/IngredientsWithTagDialogFragment.java

```diff
@@ -76,6 +79,11 @@ public static IngredientsWithTagDialogFragment newInstance(Product product, Anal
             if (showIngredients != null) {
                 args.putSerializable("ingredients", getMatchingIngredientsText(product, showIngredients.split(":")));
             }
+            Optional<LinkedHashMap<String, String>> ambiguousIngredient = product.getIngredients().stream()
```

Comment:
Please use the generic type `Optional<Map<String, String>>`.

Reply:
Modified by obtaining the list of ingredients by mapping and filtering